source/lac/trilinos_sparse_matrix.cc

```diff
@@ -2310,6 +2310,15 @@ namespace TrilinosWrappers
     return ((sizeof(TrilinosScalar)+sizeof(TrilinosWrappers::types::int_type))*
             matrix->NumMyNonzeros() + sizeof(int)*local_size() + static_memory);
   }
+
+  MPI_Comm SparseMatrix::get_mpi_communicator () const
+  {
+    const Epetra_MpiComm *comm = dynamic_cast<const Epetra_MpiComm *>(&range_partitioner().Comm());
+    if (comm)
+      return comm->Comm();
+    else
+      return MPI_COMM_WORLD;
+  }
```

Comment:
My concern was that Epetra_MpiComm.h is not available on a Trilinos without MPI, so trying to cast to Epetra_MpiComm will lead to a compile error. I think this is the reason for the code in Epetra_Vector. (Even though I admit that we should not pass a const reference to the MPI_Comm but rather return MPI_Comm by value as done here.

By the way, this reminds me that I should check the release branch with a Trilinos without MPI.